src/overlays/actors/ovl_En_Bom_Bowl_Man/z_en_bom_bowl_man.c

```diff
@@ -21,43 +50,470 @@ const ActorInit En_Bom_Bowl_Man_InitVars = {
+        //  HOW'S THE FIRST WALL DOING?
+        osSyncPrintf(VT_FGCOL(RED) "☆ 壁１の状態どう？ ☆ %d\n" VT_RST, this->wallStatus[0]);
+        //  HOW'S THE SECOND WALL DOING?
+        osSyncPrintf(VT_FGCOL(RED) "☆ 壁２の状態どう？ ☆ %d\n" VT_RST, this->wallStatus[1]);
+        // HOLE INFORMATION
+        osSyncPrintf(VT_FGCOL(RED) "☆ 穴情報\t     ☆ %d\n" VT_RST, this->bowlPit->status);
+        osSyncPrintf("\n\n");
+    }
+
+    this->gameResult = 0;
+
+    if (this->bowlPit != 0) {
+        if ((this->wallStatus[0] != 1) && (this->wallStatus[1] != 1) && (this->bowlPit->status == 2)) {
+            this->gameResult = 1; // Won
+            this->bowlPit->status = 0;
+            // Center HIT!
+            osSyncPrintf(VT_FGCOL(PURPLE) "☆☆☆☆☆ 中央ＨＩＴ！！！！ ☆☆☆☆☆ \n" VT_RST);
+        }
+        if ((globalCtx->bombchuBowlingAmmo == -1) && (globalCtx->actorCtx.actorList[3].length == 0) &&
+            (this->bowlPit->status == 0) && (this->wallStatus[0] != 1) && (this->wallStatus[1] != 1)) {
+            this->gameResult = 2; // Lost
+            // Bombchu lost
+            osSyncPrintf(VT_FGCOL(PURPLE) "☆☆☆☆☆ ボムチュウ消化 ☆☆☆☆☆ \n" VT_RST);
+        }
+    }
+
+    if (this->gameResult != 0) { // won or lost
+        this->actor.textId = 0x1A;
+        this->dialogState = 4;
+        this->minigamePlayStatus = 0;
+        if ((this->exItem != NULL) && (this->exItem->actor.update != 0)) {
+            this->exItem->unk_160 = 1;
+            this->exItem = NULL;
+        }
+        globalCtx->bombchuBowlingAmmo = 0;
+        this->playingAgain = true;
+        func_8010B680(globalCtx, this->actor.textId, NULL);
+        if (this->gameResult == 2) {
+            func_8002DF54(globalCtx, NULL, 8);
+        }
+        this->actionFunc = EnBomBowlMan_HandlePlayChoice;
+    } else {
+        if (func_8002F194(&this->actor, globalCtx)) {
+            if (this->minigamePlayStatus == 0) {
+                this->actionFunc = EnBomBowlMan_HandlePlayChoice;
+            } else {
+                this->actionFunc = func_809C41FC;
+            }
+        } else {
+            yawDiff = ABS((s16)(this->actor.yawTowardsLink - this->actor.shape.rot.y));
+            if (!(this->actor.xzDistFromLink > 120.0f) && (yawDiff < 0x4300)) {
+                func_8002F2CC(&this->actor, globalCtx, 120.0f);
+            }
+        }
+    }
+}
+
+void EnBomBowlMan_HandlePlayChoice(EnBomBowlMan* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+
+    if ((func_8010BDBC(&globalCtx->msgCtx) == this->dialogState) && (func_80106BC8(globalCtx) != 0)) {
+        func_80106CCC(globalCtx);
+
+        switch (globalCtx->msgCtx.choiceIndex) {
+            case 0: // Yes
+                if (gSaveContext.rupees >= 30) {
+                    Rupees_ChangeBy(-30);
+                    this->wallStatus[1] = 0;
+                    this->minigamePlayStatus = 1;
+                    this->wallStatus[0] = this->wallStatus[1];
```

Comment:
Could this match? Might have to play with the order a bit
```suggestion
                    this->wallStatus[1] = this->wallStatus[1] = 0;
                    this->minigamePlayStatus = 1;
```